### PR TITLE
[Project 05] Justin Chandler - Vertex Shader & Transformations

### PR DESCRIPTION
**Pull request by Justin9four**
screen shot 2019-02-20 at 9 28 49 pm
This is my completed Project 5. It was a challenge to figure out how to implement the transformations, as I experienced many challenges along the way. Most of my mistakes were simple mistakes that were easily fixed, once I realized where I had messed up.

### CODE REVIEW

File: definitions.h [hunk 1]
```diff
@@ -239,6 +239,37 @@ class Attributes
                          // double:
                          // color rgb, UV coordinates, normal vectors, fog coordinates, ECT.
         void* ptrImg;    // for the imgBuffer attribute, but may be re-purposed
+        //Matrix matrix;
```
Comment (jeremyc2) [inline definitions.h]: This sure makes it a lot faster to add a ton of attributes as once

File: pipeline.cpp [hunk 2]
```diff
@@ -153,6 +153,10 @@ void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes*
                                                           attrs[1].attr[0],attrs[2].attr[0]);
                 interpolatedAttribs.attr[1] = z * interoplate(y, x, triangle, attrs[0].attr[1],
                                                           attrs[1].attr[1],attrs[2].attr[1]);
+                interpolatedAttribs.attr[2] = z * interoplate(y, x, triangle, attrs[0].attr[2],
```
Comment (jeremyc2) [inline pipeline.cpp]: You should just keep track of how many attributes you have so you can do this in a loop depending on your number of attributes

File: coursefunctions.h [hunk 3]
```diff
@@ -449,43 +449,48 @@ void TestVertexShader(Buffer2D<PIXEL> & target)
...
-		DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
+        colorUniforms.addAttributes(100, 50);
+	DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
 
         /***********************************
          * SCALE (scale by a factor of 0.5)
          ***********************************/
         // Your scaling code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
-
+        colorUniforms.addAttributes(0,0,0,0,0.5,0.5,0);
```
Comment (wyattlarson) [inline coursefunctions.h]: I like how fast this is. I have like 10 lines of code for this one. Pretty smart. It makes it a bit less readable for reviewers like me but that doesn't really matter I guess!

File: pipeline.cpp (hunk 2 above)
Comment (wyattlarson) [inline pipeline.cpp]: I didn't do this and I don't quite understand this code you wrote either but I'm sure its important haha

**Pull request merged**